Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+            matched = target_path.match(owned_path)
+        elif owned_path.startswith("/"):  # ownership of specific directory: glob A against B
+            matched = target_path.match(owned_path[1:])
+        else:  # ownership of all relative directories: find non-wildcard portions of B in A, glob the remainders
+            asterisk = owned_path.find("*")
+            question = owned_path.find("?")
+            if asterisk > -1 and question > -1:
+                first_wildcard_index = min(asterisk, question)
+            else:  # avoid not-found index
+                first_wildcard_index = max(asterisk, question)
+            separator_indices = [index.start() for index in re.finditer(pattern="/", string=owned_path)]
+            pre_wildcard_separator_index = 0
+            for s_index in separator_indices:
+                if s_index < first_wildcard_index:
+                    pre_wildcard_separator_index = s_index
+                else:  # remainder are all greater
+                    break
+
+            # separate non-wildcard-containing path from remainder
+            pre_wildcard_owned = owned_path[:pre_wildcard_separator_index]
+            wildcard_with_remainder_owned = owned_path[pre_wildcard_separator_index+1:]
+
+            # find substrings of initial portion of B within A
+            target_str = str(target_path)
+            pre_wildcard_target_end_indices = [index.end() for index in
+                                               re.finditer(pattern=pre_wildcard_owned, string=target_str)]
+            # glob remainders of A against remainder of B
+            for target_index in pre_wildcard_target_end_indices:  # may be multiple substring matches within target
+                target_remainder = target_str[target_index:]
+                if pathlib.PurePosixPath(target_remainder).match(wildcard_with_remainder_owned):
+                    matched = True
+                    break  # exit early on success
+    else:  # simple path matching
+        if owned_path.startswith("/"):  # ownership of specific directory: verify if A exists inside B
+            try:  # is_relative_to introduced in Python 3.9
+                target_path.relative_to(owned_path)
+                matched = True
+            except ValueError:
+                matched = False
+        else:  # ownership of all relative directories: verify if B is a substring of A
+            matched = owned_path in str(target_path)
+    return matched
+
+
+def _pretty_print_success(print_fn, found_codeowners_path, matched_path, owner_aliases):
+    """
+    :param print_fn: function to call when logging strings
+    :param found_codeowners_path: verified path to a GitHub CODEOWNERS file
+    :param matched_path: first part of an entry matched in the CODEOWNERS file
+    :param owner_aliases: second part of an entry in a CODEOWNERS file
+    """
+    print_fn(f"Matched '{matched_path}' in file {found_codeowners_path}")
```

Comment:
I think the formatting could be simplified/improved. Maybe something as simple as:
`Github Codeowner: @o3de/sig-testing-reviewers`

Reply:
If the user wants simple output, they don't need to send the pretty_print argument. Then it will only output the alias(es)

Reply:
I'm happy to make the second line more accurate and explain it as a GitHub Codeowner, but I don't think this line should be removed. In the case where the user wants a different pattern to match a file, ine 165 makes it easy to know what file to search in and what line to search for.

Reply:
> If the user wants simple output, they don't need to send the pretty_print argument. Then it will only output the alias(es)

The default for AR will be pretty print though right? I just think it's a more difficult to parse at a glance with full sentences.

Reply:
Correct, the default for PyTest logs is set to the friendly sentence.

I personally disagree that it is more difficult to parse without stating the motivation/intent of using code ownership, but am interested in hearing more opinions about what gets logged here.

Reply:
From a user perspective who is unaware of this tool, I don't think they will read much when looking through the logs. Having an explicit simple line I feel is more direct and to the point. Maybe we can have both?

Reply:
Could you clarify what precisely you're recommending? I'm imagining it as:
```
Matched '/Code/Editor/' in file .github\CODEOWNERS
GitHub Codeowners: @o3de/sig-content-maintainers
```
Versus the current:
```
Matched '/Code/Editor/' in file .github\CODEOWNERS
For additional support please reach out to: @o3de/sig-content-maintainers
```

Reply:
Yeah I think your suggestion looks good. Maybe even a new line to help it stand out?

Reply:
I still prefer the intent and action suggested by the second/current message, but I'll look to additional feedback by more reviews.